doc/api.md

```diff
@@ -17,339 +17,375 @@ This Document documents the types introduced by the GMP CRDs to be consumed by u
 > Note this document is generated from code comments. When contributing a change to this document please do so by changing the code comments.
 
 ## Table of Contents
-* [AlertingSpec](#alertingspec)
-* [AlertmanagerEndpoints](#alertmanagerendpoints)
-* [Auth](#auth)
-* [Authorization](#authorization)
-* [BasicAuth](#basicauth)
 * [ClusterPodMonitoring](#clusterpodmonitoring)
```

Comment:
Hm, this is bit brutal consequence. It is definitely easier for users to read this when structs are sorted lex. I guess we could do some post processing too, but maybe there is a way to contribute/configure generator? 🤔

Reply:
The tool we use here was deleted by the maintainers. We currently use `po-docgen`, which was originally developed by prometheus-operator. They have since switched to `gen-crd-api-reference-docs`. However, if you go to the `gen-crd-api-reference-docs` README, it states that it will not be maintained anymore either, with 3 possible alternatives. We should switch to a different tool but this is more work that I'm not getting myself into yet.

Reply:
Address in #756, which this branch was rebased off of. Please review that PR first!